tests/helpers/perf_results_helper.rb

```diff
@@ -757,5 +760,73 @@ def extract_puppetserver_metrics_from_json(file)
 
     row
   end
+
+  # Execute the specified SQL query via BigQuery
+  #
+  # @author Bill Claytor
+  #
+  # @raise [StandardError] If the GOOGLE_APPLICATION_CREDENTIALS environment variable is not set
+  #
+  # @return [Array] The array of versions
+  #
+  # @example
+  #   versions_array = baseline_versions
+  #
+  def query_bigquery(sql)
+    error = "The GOOGLE_APPLICATION_CREDENTIALS environment variable must be set with the location of the JSON key file"
+    raise error if ENV["GOOGLE_APPLICATION_CREDENTIALS"].nil? || ENV["GOOGLE_APPLICATION_CREDENTIALS"].empty?
+
+    bigquery = Google::Cloud::Bigquery.new project: BIGQUERY_PROJECT
+    data = bigquery.query sql
+    logger.error("Cannot find result that matches query: #{sql}") if data.empty?
+    data
+  end
+
+  # Retrieve the list of baseline PE versions from BigQuery
+  #
+  # @author Bill Claytor
+  #
+  # @raise [StandardError] If the query returns no data
+  #
+  # @return [Array] The array of versions
+  #
+  # @example
+  #   versions_array = baseline_versions
+  #
+  def baseline_versions
+    sql = "SELECT DISTINCT pe_build_number FROM `perf-metrics.perf_metrics.atop_metrics` " \
+          "WHERE pe_build_number IS NOT NULL " \
+          "ORDER BY pe_build_number"
+    data = query_bigquery(sql)
+    raise "Error: cannot find result that matches query: #{sql}" if data.empty?
```

Comment:
Should probably say more about where the query was sent... like the big query project name, and that it was big query.

Reply:
Good call; updated error messages.